### DISCUSSION TITLE
Get exception messages as C# object/JSON

### DISCUSSION THREAD

**Discussion by q34hfq34fq5374rqhdf0vih3435dd23234wkklo** [fluentassertions/fluentassertions#2666] [Q&A] [answered]
Hi,

Is there a way to get the failed assertions from the exception message as a C# object/JSON string? I would like to iterate over the failed assertions and do some actions for my objects. If my use case cannot be implemented with this library, are there any alternatives that you know of?

**Example use case:**
```cs
// firstObject.thing = null and secondObject.thing = 123456
firstObject.Should().BeEquivalentTo(secondObject, options => options.Excluding(o => o.otherThing)...);
// Expected property firstObject.thing to be 123456, but found <null>.
```
I would like to iterate through the failed assertions and set `firstObject.thing` to the expected value of 123456.

Thank you

**Comment by dennisdoomen** [comment, marked as answer]
No, the only thing you'll get is the framework-specific exception, e.g. `XUnitException`.

**Reply by q34hfq34fq5374rqhdf0vih3435dd23234wkklo** [reply]
Ended up using FluentAssertions with: https://github.com/replaysMike/AnyDiff

**Comment by dennisdoomen** [comment]
Huh. If you need to see which exact mismatches there are, you already get that information as part of the `XUnitException`.

**Reply by q34hfq34fq5374rqhdf0vih3435dd23234wkklo** [reply]
Hi,

I am not using this for testing purposes but to do a delta on two objects. The mismatches are in the exceptions as strings, but I did not want to parse the strings to get data.

I use the AnyDiff library to get the mismatch (which has the values of each object, the delta, and other info) in a C# object so I can do an action on one of my objects based on the values that are different. I still like the output of FluentAssertions and use it as a double check (ie. both libraries should output the same count for mismatches)

My use case is probably not the intention of this library, but I do still find it useful.

Thanks 🙂
